### PR TITLE
AC: Remove legacy AC from folders permissions API

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Remove legacy AC from folders permissions API

Next PR remove UpdateACL interface method

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Related https://github.com/grafana/grafana/issues/68987
Related https://github.com/grafana/grafana/issues/68076

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by zserge — APPROVED** [review]

**Review by papagian — APPROVED** [review]
I assume that `hs.DashboardService.UpdateDashboardACL()` was only for legacy AC?

**Comment by Jguer** [comment]
yes @papagian , with AC now always enabled we can start removing these https://github.com/grafana/grafana/issues/68076

**Review by kalleep — APPROVED** [review]

**Review by IevaVasiljeva — APPROVED** [review]

File: pkg/api/folder_permission.go [hunk 1]
```diff
@@ -128,54 +127,31 @@ func (hs *HTTPServer) UpdateFolderPermissions(c *contextmodel.ReqContext) respon
...
-		if err != nil {
-			return response.Error(500, "Error while checking dashboard permissions", err)
-		}
-		if err := hs.updateDashboardAccessControl(c.Req.Context(), c.OrgID, folder.UID, true, items, old); err != nil {
-			return response.Error(500, "Failed to create permission", err)
-		}
-		return response.Success("Dashboard permissions updated")
+	old, err := g.GetACL()
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Error while checking dashboard permissions", err)
```
Comment (IevaVasiljeva) [inline pkg/api/folder_permission.go:149]: nit: I know you didn't introduce it, but could you please update this to
```suggestion
		return response.Error(http.StatusInternalServerError, "Error while checking folder permissions", err)
```

**Review by IevaVasiljeva — COMMENTED** [review]
Just a quick comment on one of the error messages.

**Pull request merged**